jwt/src/test/java/me/uport/sdk/jwt/TimestampTests.kt

```diff
@@ -0,0 +1,156 @@
+package me.uport.sdk.jwt
+
+import assertk.assertThat
+import assertk.assertions.hasMessage
+import assertk.assertions.isEqualTo
+import assertk.assertions.isInstanceOf
+import io.mockk.coEvery
+import io.mockk.mockk
+import io.mockk.spyk
+import kotlinx.coroutines.runBlocking
+import me.uport.sdk.ethrdid.EthrDIDResolver
+import me.uport.sdk.jsonrpc.JsonRPC
+import me.uport.sdk.jwt.test.EthrDIDTestHelpers.Companion.mockDocForAddress
+import me.uport.sdk.testhelpers.TestTimeProvider
+import me.uport.sdk.testhelpers.coAssert
+import me.uport.sdk.universaldid.UniversalDID
+import org.junit.Before
+import org.junit.Test
+
+class TimestampTests {
+
+    companion object {
+
+        //mock timestamps
+
+        private const val NOW: Long = 1_500_000L //milliseconds
+        private const val PAST: Long = 1_100L //seconds
+        private const val FUTURE: Long = 1_900L //seconds
+    }
+
+    @Before
+    fun `mock DID documents before every test`() {
+        val jrpc = mockk<JsonRPC>()
+        val resolver = spyk(EthrDIDResolver(jrpc))
+        coEvery {
+            resolver.resolve(any())
+        } returns mockDocForAddress("0xcf03dd0a894ef79cb5b601a43c4b25e3ae4c67ed")
+
+        UniversalDID.clearResolvers()
+        UniversalDID.registerResolver(resolver)
+    }
+
+
+
+
+    @Test
+    fun `pass when nbf exists in the past`() = runBlocking {
+        val tested = JWTTools(TestTimeProvider(NOW))
+
+        val jwt =
+            """eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJuYmYiOjExMDAsImlzcyI6IjB4Y2YwM2RkMGE4OTRlZjc5Y2I1YjYwMWE0M2M0YjI1ZTNhZTRjNjdlZCJ9.w14TBtuK2k65tGYk1QmkehfQt3CBwTS0h23yKIliZZNB3wLgidyNpm8hIr04PLv4j-ayDlOLCZL73fGkd6YIJw"""
+        val (_, decoded, _) = tested.decodeRaw(jwt)
+        assertThat(decoded["nbf"]).isEqualTo(PAST)
+        assertThat(decoded.containsKey("iat")).isEqualTo(false)
+
+        coAssert {
+            tested.verify(jwt)
+        }.doesNotThrowAnyException()
+    }
+
+    @Test
+    fun `pass when nbf exists in the past and iat in the future`() = runBlocking {
```

Comment:
Similar to the comment above: I don't think it's necessary to make explicit assertions about it because they can be consolidated under the case that `iat` should be ignored during verification if `nbf` is present.

Reply:
> iat should be ignored during verification if nbf is present.

right. that is exactly the point of this test.
Verification would fail if `iat` were not ignored, that's why it is so explicit.